### PR TITLE
Clean up empty autoprovisioned node groups

### PR DESCRIPTION
**Pull request by mwielgus**

### CODE REVIEW

File: cluster-autoscaler/core/static_autoscaler.go [hunk 1]
```diff
@@ -287,6 +287,10 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 		if !calculateUnneededOnly {
 			glog.V(4).Infof("Starting scale down")
 
+			if a.AutoscalingContext.AutoscalingOptions.NodeAutoprovisioningEnabled {
```
Comment (aleksandra-malinowska) [inline cluster-autoscaler/core/static_autoscaler.go]: ".AutoscalingOptions" not necessary

Comment (aleksandra-malinowska) [inline cluster-autoscaler/core/static_autoscaler.go]: I'm not sure if this should be behind !calculateUnneededOnly check. How expensive is adding/deleting a nodegroup?

Comment (mwielgus) [reply cluster-autoscaler/core/static_autoscaler.go]: Done.

Comment (mwielgus) [reply cluster-autoscaler/core/static_autoscaler.go]: This is not about expensive operations but about not deleting node pools too early or under conditions that we would not even allow to delete a node. I added a comment to explain it better.

**Pull request merged**